File: backend/src/main/scala/model/Target.scala
package model

import scala.slick.driver.MySQLDriver.simple._

case class Target(targetId: Int,
                  eventId: String,
                  userPhoneNumber: String)

class Targets(tag: Tag) extends Table[Target](tag, "targets") {
  def targetId = column[Int]("target_id", O.PrimaryKey, O.AutoInc)
  def eventId = column[String]("event_id", O.NotNull)
  def userPhoneNumber = column[String]("user_phone_number", O.NotNull)

  def eventIndex = index("tgt_event_id_index", eventId, unique = false)
  def userPhoneNumberIndex = index("tgt_phone_number_index", userPhoneNumber, unique = false)
  def user = foreignKey("tgt_user_fk", userPhoneNumber, TableQuery[Users])(_.phoneNumber)
  def event = foreignKey("tgt_event_fk", eventId, TableQuery[Events])(_.eventId)

  def * = (targetId, eventId, userPhoneNumber) <> (Target.tupled, Target.unapply)
}

File: server/code/model/target.py
from base import Base, Ninja_Model_Mixin
from sqlalchemy import Column, String, ForeignKey, Integer
from sqlalchemy.orm import relationship, backref
from model.event import Event


class Target(Base, Ninja_Model_Mixin):
    __tablename__ = 'targets'
    target_id = Column('target_id', Integer, primary_key = True)
    event_id = Column('event_id', String(254), ForeignKey("events.event_id"), nullable = False)
    user_phone_number = Column('user_phone_number', String(254), ForeignKey("authentication.phonenumber"), nullable = False)

    event = relationship("Event", backref = "target")
